extensions/2.0/Vendor/FB_material_variants/README.md

```diff
@@ -0,0 +1,98 @@
+
+## The variant mapping as glTF JSON
+
+A snippet of a mesh implementing shoe holes which uses this extension might look like:
+
+```javascript
+"meshes": [
+    {
+        "name": "shoelace_hole",
+        "primitives": [
+            {
+                "attributes": {
+                "POSITION": 0,
+                "NORMAL": 1,
+                },
+                "indices": 2,
+                "material": 0,
+                "extensions": {
+                    "FB_material_variants" : {
+                        "mapping": [
+                            {
+                                "tags": [ "sneaker_yellow", "sneaker_orange" ],
+                                "material": 7,
+                            },
+                            {
+                                "tags": [ "sneaker_red" ],
+                                "material": 8,
+                            },
+                            {
+                                "tags": [ "sneaker_black" ],
+                                "material": 9,
+                            },
+                        ],
+                    }
+                }
+            },
+            // ... more primitives ...
+        ]
+    },
+    // ... more meshes ...
+]
+```
+
+The tag-based approach allows highly selective material switching, e.g. changing the colour of shoe laces only, or wholesale changes across diverse geometry, by reusing the same tags in switches across many distinct mesh primitives.
+
+In the typical case, many tags will be active, e.g. `['sneaker_red', 'laces_gold']`.
+
+Composition also works well with tags: if a scene were built from several different variational models, each with their own set of tags, then it's easy to imagine an API call on the entire scene that takes a set of tags and passes them on to each constituent model, recursively.
+
+## Interaction with existing glTF functionality
+
+The primary purpose of this extension is to simply formalise the idea of configurability, so that members of the ecosystem can meaningfully communicate: those who produce asset and those who ingest them, and all the tooling inbetween.
+
+Beyond that, it's also a compression scheme: a way to compactly represent multiple assets that have a lot in common (e.g. most geometry, some textures). In this respect, it's most meaningful for the self-contained binary format GLB, where the asset file contains all its data up-front, both that which is shared between variants and that which differs.
+
+If we also allow external references, i.e. images and buffers to source from URLs,then we can additionally come up with sophisticated mechanisms where geometry and textures are loaded separately, dynamically. There's pragmatic pros and cons to going down this route, all beyond the scope of this document. Each application will make its own decisions there; meawhile, the extension proposed works well regardless.
```

Comment:
These two paragraphs (`Beyond that, ...` and `If we also...`) seem to contradict one another, and may be confusing as-is. I think you could effectively condense them to something like:

> As a secondary effect, material variants allow multiple assets — with shared geometry but different materials — to be stored more compactly. When using external URIs as references to textures, applications may (optionally) process geometry only once and lazily request texture assets only when needed for a particular variant.

***

On this section...

> If we also allow external references, i.e. images and buffers to source from URLs,then we can additionally come up with sophisticated mechanisms where geometry and textures are loaded separately, dynamically.

... I would consider this a certainty rather than an "if"; incrementally loading parts of a glTF asset is already possible and supported in engines like three.js. It's probably out of scope for this document, as you say, but I would strongly expect (though not require) that applications will take advantage of it.

Reply:
Thanks for the suggestions here @donmccurdy, I've gone ahead and condensed the language of the non-normative sections as per your proposal